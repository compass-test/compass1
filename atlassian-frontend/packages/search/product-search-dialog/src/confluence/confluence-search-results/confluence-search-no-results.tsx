import React, { useCallback } from 'react';
import { withAnalytics } from '../../common/analytics';
import {
  onAdvancedSearchSelected,
  onFiltersCleared,
} from '../../common/analytics/events';
import { NoResults, NoResultsProps } from '../../common/no-results';
import { useFeatures } from '../confluence-features';
import { useFilterContext } from '../filter-context';

interface Props {
  filtersCleared: () => void;
}

export const NoResultsWithHooks = (props: Props & NoResultsProps) => {
  const { isMultiSite } = useFeatures();
  const filterContext = useFilterContext();
  const enabledSpaceFilters = filterContext.spaceFilters.availableFilters.filter(
    (f) => f.isChecked,
  );
  const enabledPeopleFilters = filterContext.peopleFilters.availableFilters.filter(
    (f) => f.isChecked,
  );
  const enabledSiteFilters = filterContext.siteFilters.availableFilters.filter(
    (f) => f.isChecked,
  );

  const hasFilters =
    enabledSpaceFilters.length +
      enabledPeopleFilters.length +
      enabledSiteFilters.length >
    0;

  const clearSpaceFilters = filterContext.spaceFilters.clearFilter;
  const clearPeopleFilters = filterContext.peopleFilters.clearFilter;
  const clearSiteFilters = filterContext.siteFilters.clearFilter;

  const { filtersCleared } = props;

  const clearAllFilters = useCallback(() => {
    filtersCleared();

    clearSpaceFilters();
    clearPeopleFilters();
    clearSiteFilters();
  }, [clearSpaceFilters, clearPeopleFilters, clearSiteFilters, filtersCleared]);

  return (
    <NoResults
      {...props}
      hasFilters={hasFilters}
      clearFilters={clearAllFilters}
      isMultiSite={isMultiSite}
    />
  );
};

export default withAnalytics({
  advancedSearchSelected: onAdvancedSearchSelected,
  filtersCleared: onFiltersCleared,
})(NoResultsWithHooks);
