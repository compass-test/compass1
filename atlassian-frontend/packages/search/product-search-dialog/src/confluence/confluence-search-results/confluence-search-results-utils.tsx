import {
  ConfItemResult,
  ConfSpaceResult,
  ConfPeopleResult,
  ContentType,
  ConfluencePage,
  ConfluenceBlogpost,
} from '../clients/response-types';

import { getTrigger, isNewTab } from '../../common/analytics';
import {
  ResultSectionAnalyticData,
  AdvancedSearchAnalyticsProps,
  AdvancedSearchLinkSubjectId,
} from '../../common/analytics/events';
import React from 'react';
import { B200 } from '@atlaskit/theme/colors';
import BlogIcon16 from '@atlaskit/icon-object/glyph/blog/16';
import PageIcon16 from '@atlaskit/icon-object/glyph/page/16';
import BlogIcon24 from '@atlaskit/icon-object/glyph/blog/24';
import PageIcon24 from '@atlaskit/icon-object/glyph/page/24';

export const getResultsDetailsAnalyticData = (
  itemResults: ConfItemResult[],
  spaceResults: ConfSpaceResult[],
  peopleResults: ConfPeopleResult[],
  isPreQuery: boolean,
) => {
  const resultAnalyticData: ResultSectionAnalyticData[] = [];
  itemResults.length > 0 &&
    resultAnalyticData.push({
      results: itemResults.map((item) => ({
        containerId: item.containerId,
        isRecentResult: item.isRecentResult,
        resultContentId: item.resultId,
        resultType: item.contentType,
      })),
      sectionId: 'confluence-item',
    });

  spaceResults.length > 0 &&
    resultAnalyticData.push({
      results: spaceResults.map((item) => ({
        containerId: 'UNAVAILABLE',
        isRecentResult: isPreQuery,
        resultContentId: item.resultId,
        resultType: item.contentType,
      })),
      sectionId: 'confluence-space',
    });

  peopleResults.length > 0 &&
    resultAnalyticData.push({
      results: peopleResults.map((item) => ({
        containerId: 'UNAVAILABLE',
        isRecentResult: isPreQuery,
        resultContentId: item.resultId,
        resultType: item.contentType,
      })),
      sectionId: 'confluence-people',
    });

  return resultAnalyticData;
};

export const firePeopleSearchSelected = (
  e: React.MouseEvent<HTMLSpanElement, MouseEvent>,
  advancedSearchSelected: (props: AdvancedSearchAnalyticsProps) => any,
) => {
  advancedSearchSelected({
    trigger: getTrigger(e),
    actionSubjectId: AdvancedSearchLinkSubjectId.CONFLUENCE_PEOPLE,
    isLoading: false,
    newTab: isNewTab(e),
  });
};

export const getAvatarForConfluenceObjectResult = (
  contentType: ContentType,
  name: string,
  largerIcons?: boolean,
) => {
  const props = {
    primaryColor: B200,
    label: name,
  };

  if (contentType === ConfluencePage) {
    return largerIcons ? <PageIcon24 {...props} /> : <PageIcon16 {...props} />;
  } else if (contentType === ConfluenceBlogpost) {
    return largerIcons ? <BlogIcon24 {...props} /> : <BlogIcon16 {...props} />;
  }

  return <></>;
};
