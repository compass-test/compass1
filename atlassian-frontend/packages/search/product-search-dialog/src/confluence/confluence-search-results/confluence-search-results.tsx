import React from 'react';
import { LinkComponent } from '@atlassian/search-dialog';
import {
  ConfItemResults,
  ConfSpaceResults,
  ConfPeopleResults,
} from '../clients/response-types';
import { ConfluencePreQuery } from './pre-query';
import { ConfluencePostQuery } from './post-query';
import { LoadingSpinner } from '../../common/loading-spinner';
import { usePrimarySiteConfluenceAdvancedSearchUrlFactory } from '../confluence-utils/confluence-url-utils';
import NoResults from './confluence-search-no-results';
import { NoPreQueryResults } from '../../common/no-results';

interface Props {
  query: string;
  isPreQuery: boolean;
  linkComponent?: LinkComponent;
  formatDate?: (lastModified: string) => JSX.Element;
  items: {
    isLoading: boolean;
    results: ConfItemResults | null;
  };
  spaces: {
    isLoading: boolean;
    results: ConfSpaceResults | null;
  };
  people: {
    isLoading: boolean;
    results: ConfPeopleResults | null;
  };
  isBelowTakeoverWidth: boolean;
  showFilters?: boolean;
}

interface ContextProps {
  advancedSearchUrl: string;
}

export enum ResultState {
  PreQueryLoading,
  PostQueryLoading,

  PreQueryResults,
  PostQueryResults,
}

export const getCurrentState = (props: Props) => {
  const { items, spaces, isPreQuery } = props;
  const isLoading = items.isLoading || spaces.isLoading;
  if (isLoading) {
    return isPreQuery
      ? ResultState.PreQueryLoading
      : ResultState.PostQueryLoading;
  } else {
    return isPreQuery
      ? ResultState.PreQueryResults
      : ResultState.PostQueryResults;
  }
};

export class SearchResults extends React.Component<Props & ContextProps> {
  static getFetchedResultsCount(props: Props) {
    const { items, spaces, people } = props;

    const itemLength = (items.results && items.results.items.length) || 0;
    const spacesLength = (spaces.results && spaces.results.items.length) || 0;
    const peopleLength = (people.results && people.results.items.length) || 0;

    return {
      itemLength,
      spacesLength,
      peopleLength,
      totalCount: itemLength + spacesLength + peopleLength,
    };
  }

  static hasResults(props: Props) {
    return SearchResults.getFetchedResultsCount(props).totalCount > 0;
  }

  render() {
    const {
      query,
      items,
      spaces,
      people,
      linkComponent,
      formatDate,
      isBelowTakeoverWidth,
      showFilters = false,
      advancedSearchUrl,
    } = this.props;

    const currentState = getCurrentState(this.props);
    const hasResults = SearchResults.hasResults(this.props);

    let results;
    switch (currentState) {
      case ResultState.PreQueryLoading: {
        results = <LoadingSpinner />;
        break;
      }
      case ResultState.PostQueryLoading: {
        results = (
          <>
            {hasResults ? (
              <ConfluencePostQuery
                query={query}
                items={items}
                spaces={spaces}
                people={people}
                linkComponent={linkComponent}
                formatDate={formatDate}
                isLoading
                screenType="cachedResults"
                isCollapsed={isBelowTakeoverWidth || showFilters}
              />
            ) : null}
            <LoadingSpinner />
          </>
        );
        break;
      }
      case ResultState.PreQueryResults: {
        results = hasResults ? (
          <ConfluencePreQuery
            items={items}
            spaces={spaces}
            people={people}
            linkComponent={linkComponent}
            isLoading={false}
            isCollapsed={isBelowTakeoverWidth}
          />
        ) : (
          <NoPreQueryResults />
        );

        break;
      }
      case ResultState.PostQueryResults: {
        results = (
          <>
            {hasResults ? (
              <ConfluencePostQuery
                query={query}
                items={items}
                spaces={spaces}
                people={people}
                linkComponent={linkComponent}
                isLoading={false}
                formatDate={formatDate}
                screenType="postQuerySearchResults"
                isCollapsed={isBelowTakeoverWidth || showFilters}
              />
            ) : (
              <NoResults
                linkComponent={linkComponent}
                advancedSearchUrl={advancedSearchUrl}
              />
            )}
          </>
        );
        break;
      }
      default: {
        // This is to ensure the switch is exhaustive
        const error: never = currentState;
        throw new Error(error);
      }
    }
    return results;
  }
}

export default (props: Props) => {
  const advancedSearchUrl = usePrimarySiteConfluenceAdvancedSearchUrlFactory()(
    props.query,
  );
  return <SearchResults {...props} advancedSearchUrl={advancedSearchUrl} />;
};
