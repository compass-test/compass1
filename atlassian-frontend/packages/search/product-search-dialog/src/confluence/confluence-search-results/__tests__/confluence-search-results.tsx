import React from 'react';
import { shallow } from 'enzyme';
import {
  createPageBlogAttachmentResults,
  createSpaceResponse,
  createPeopleResults,
  enableDeterministicResponses,
} from '../../../__tests__/__fixtures__/mock-search-results';
import { SearchResults } from '../confluence-search-results';
import { ConfluencePreQuery } from '../pre-query';
import { ConfluencePostQuery } from '../post-query';
import { LoadingSpinner } from '../../../common/loading-spinner';
import { FilterPane } from '../../filter';
import NoResults from '../confluence-search-no-results';
import { DefaultFeatures } from '../../confluence-features';

jest.mock('../pre-query');
jest.mock('../post-query');
jest.mock('../../../common/loading-spinner', () => ({
  LoadingSpinner: () => 'div',
}));
jest.mock('@atlaskit/width-detector', () => ({
  WidthObserver: () => <div />,
}));
jest.mock('../../filter');

describe('<SearchResults />', () => {
  const createResults = () => ({
    items: {
      isLoading: false,
      results: createPageBlogAttachmentResults(1),
    },
    spaces: {
      isLoading: false,
      results: createSpaceResponse(1),
    },
    people: {
      isLoading: false,
      results: createPeopleResults(1),
    },
  });

  const emptyResults = {
    items: {
      isLoading: false,
      results: createPageBlogAttachmentResults(0),
    },
    spaces: {
      isLoading: false,
      results: createSpaceResponse(0),
    },
    people: {
      isLoading: false,
      results: createPeopleResults(0),
    },
  };

  const onFiltersCleared = jest.fn();

  const LoadingResultState = {
    isLoading: true,
    results: null,
  };

  const loadingProps = {
    items: LoadingResultState,
    spaces: LoadingResultState,
    people: LoadingResultState,
  };

  const baseProps = {
    query: 'query',
    advancedSearchUrl: 'some/advanced/search/url?q=query',
    enabledFilters: [],
    onSearchResultsShown: jest.fn(),
    searchSessionId: 'testSearchSeassionId',
    features: DefaultFeatures,
    isBelowTakeoverWidth: false,
    onFiltersCleared,
  };

  beforeAll(() => {
    enableDeterministicResponses();
  });

  it('render with results matches snapshot for pre query', () => {
    const wrapper = shallow(
      <SearchResults {...baseProps} {...createResults()} isPreQuery />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results matches snapshot for pre query with filter on', () => {
    const wrapper = shallow(
      <SearchResults
        {...baseProps}
        {...createResults()}
        isPreQuery
        showFilters
        isBelowTakeoverWidth={false}
      />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
    expect(wrapper.find(FilterPane).exists()).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results matches snapshot for post query', () => {
    const wrapper = shallow(
      <SearchResults {...baseProps} {...createResults()} isPreQuery={false} />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).prop('screenType')).toBe(
      'postQuerySearchResults',
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render with results matches snapshot for post query with filter on', () => {
    const wrapper = shallow(
      <SearchResults
        {...baseProps}
        {...createResults()}
        isPreQuery={false}
        showFilters
        isBelowTakeoverWidth={false}
      />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('post query with filter on & no results renders correctly', () => {
    const wrapper = shallow(
      <SearchResults
        {...baseProps}
        {...emptyResults}
        isPreQuery={false}
        showFilters
        isBelowTakeoverWidth={false}
      />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
    expect(wrapper.find(NoResults).exists()).toBe(true);
  });

  it('render with results matches snapshot for pre query loading', () => {
    const wrapper = shallow(
      <SearchResults {...baseProps} {...loadingProps} isPreQuery />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
    expect(wrapper.find(LoadingSpinner).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  describe('with no results', () => {
    let wrapper: any;
    beforeEach(() => {
      baseProps.onSearchResultsShown = jest.fn();
      wrapper = shallow(
        <SearchResults {...baseProps} {...loadingProps} isPreQuery={false} />,
      );
    });

    it('render and matches snapshot for post query loading', () => {
      expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
      expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
      expect(wrapper.find(LoadingSpinner).exists()).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('render with intermediate results matches snapshot for post query loading', () => {
    const wrapper = shallow(
      <SearchResults
        {...baseProps}
        {...loadingProps}
        items={{
          isLoading: false,
          results: createPageBlogAttachmentResults(1),
        }}
        isPreQuery={false}
      />,
    );

    expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);

    // We expect the post query screen to show here along with the spinner
    expect(wrapper.find(ConfluencePostQuery).exists()).toBe(true);
    expect(wrapper.find(ConfluencePostQuery).prop('screenType')).toBe(
      'cachedResults',
    );
    expect(wrapper.find(LoadingSpinner).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  describe('on mobile, ', () => {
    beforeAll(() => {
      Object.defineProperty(document.body, 'clientWidth', {
        value: 600,
        writable: true,
      });
    });

    it('render with results matches snapshot for pre query with filter on', () => {
      const wrapper = shallow(
        <SearchResults
          {...baseProps}
          {...createResults()}
          isPreQuery
          showFilters
          isBelowTakeoverWidth={false}
        />,
      );

      expect(wrapper.find(ConfluencePreQuery).exists()).toBe(true);
      expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
      expect(wrapper.find(FilterPane).exists()).toBe(false);
      expect(wrapper).toMatchSnapshot();
    });

    it('render with results matches snapshot for post query with filter on', () => {
      const wrapper = shallow(
        <SearchResults
          {...baseProps}
          {...createResults()}
          isPreQuery={false}
          showFilters
          isBelowTakeoverWidth
        />,
      );

      expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
      expect(wrapper.find(ConfluencePostQuery).exists()).toBe(true);
      expect(wrapper.find(FilterPane).exists()).toBe(false);
      expect(wrapper).toMatchSnapshot();
    });

    it('post query with filter on & no results renders correctly', () => {
      const wrapper = shallow(
        <SearchResults
          {...baseProps}
          {...emptyResults}
          isPreQuery={false}
          showFilters
          isBelowTakeoverWidth
        />,
      );

      expect(wrapper.find(ConfluencePreQuery).exists()).toBe(false);
      expect(wrapper.find(ConfluencePostQuery).exists()).toBe(false);
      expect(wrapper.find(NoResults).exists()).toBe(true);
      expect(wrapper.find(FilterPane).exists()).toBe(false);
    });
  });
});
