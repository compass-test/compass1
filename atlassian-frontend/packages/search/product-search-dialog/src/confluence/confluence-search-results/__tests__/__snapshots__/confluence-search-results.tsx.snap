// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> on mobile,  render with results matches snapshot for post query with filter on 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    isCollapsed={true}
    isLoading={false}
    items={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-confluence",
              "containerId": "UNAVAILABLE",
              "containerName": "Muller and Sons",
              "contentType": "confluence-page",
              "friendlyLastModified": "about 7 hours ago",
              "href": "http://baseUrl/spaces/18b58081-6d12-4e0b-b339-8f2d7faa9aa5",
              "iconClass": "aui-iconfont-page-default",
              "isRecentResult": false,
              "lastModified": "2019-07-08T02:54:38.822Z",
              "name": "modi rerum cupiditate rerum non",
              "resultId": "18b58081-6d12-4e0b-b339-8f2d7faa9aa5",
              "resultType": "confluence-object-result",
              "spaceId": "b54f4888-45ab-4761-8b95-891f984ce2c0",
            },
          ],
          "timings": 0,
          "totalSize": 99,
        },
      }
    }
    people={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-person",
              "avatarUrl": "http://lorempixel.com/640/480/people",
              "contentType": "person",
              "href": "/people/cff13b27-d1f2-4577-85cf-76fdd156f9fb",
              "mentionName": "nickname-4v3u3",
              "name": "Santina Feeney",
              "presenceMessage": "",
              "resultId": "people-cff13b27-d1f2-4577-85cf-76fdd156f9fb",
              "resultType": "person-result",
              "userId": "cff13b27-d1f2-4577-85cf-76fdd156f9fb",
            },
          ],
          "timings": 0,
        },
      }
    }
    query="query"
    screenType="postQuerySearchResults"
    spaces={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-confluence",
              "avatarUrl": "http://lorempixel.com/640/480/cats",
              "contentType": "confluence-space",
              "href": "http://baseUrl/spaces/PRODUCT",
              "id": "155c133c-18ba-4bf7-9556-ce172460805c",
              "key": "PRODUCT",
              "name": "Kuhn - Feest",
              "resultId": "space-PRODUCT",
              "resultType": "generic-container-result",
            },
          ],
          "timings": 0,
        },
      }
    }
  />
</Fragment>
`;

exports[`<SearchResults /> on mobile,  render with results matches snapshot for pre query with filter on 1`] = `
<WithAnalyticsEvents(WrappedComponent)
  isCollapsed={false}
  isLoading={false}
  items={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-confluence",
            "containerId": "UNAVAILABLE",
            "containerName": "Blanda - Dickinson",
            "contentType": "confluence-attachment",
            "friendlyLastModified": "Dec 23, 2018",
            "href": "http://baseUrl/spaces/c86fb60d-c7d1-4c2c-9cbc-8c96387134f8",
            "iconClass": "icon-file-multimedia",
            "isRecentResult": false,
            "lastModified": "2019-07-08T02:54:38.822Z",
            "name": "labore.wma",
            "resultId": "c86fb60d-c7d1-4c2c-9cbc-8c96387134f8",
            "resultType": "confluence-object-result",
            "spaceId": "b535aacc-578b-4fe3-9665-9bca9a8c2762",
          },
        ],
        "timings": 0,
        "totalSize": 99,
      },
    }
  }
  people={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-person",
            "avatarUrl": "http://lorempixel.com/640/480/people",
            "contentType": "person",
            "href": "/people/ae53c8fe-7db1-43ce-ae85-00b4880acd56",
            "mentionName": "nickname-0py2i",
            "name": "Vada Lueilwitz IV",
            "presenceMessage": "",
            "resultId": "people-ae53c8fe-7db1-43ce-ae85-00b4880acd56",
            "resultType": "person-result",
            "userId": "ae53c8fe-7db1-43ce-ae85-00b4880acd56",
          },
        ],
        "timings": 0,
      },
    }
  }
  spaces={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-confluence",
            "avatarUrl": "http://lorempixel.com/640/480/cats",
            "contentType": "confluence-space",
            "href": "http://baseUrl/spaces/BORDERS",
            "id": "385c16b5-f5a1-4528-ae79-0ec9ded29035",
            "key": "BORDERS",
            "name": "Anderson, Parker and Kilback",
            "resultId": "space-BORDERS",
            "resultType": "generic-container-result",
          },
        ],
        "timings": 0,
      },
    }
  }
/>
`;

exports[`<SearchResults /> render with intermediate results matches snapshot for post query loading 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    isCollapsed={false}
    isLoading={true}
    items={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-confluence",
              "containerId": "UNAVAILABLE",
              "containerName": "Olson - Lakin",
              "contentType": "confluence-attachment",
              "friendlyLastModified": "Jan 23, 2018",
              "href": "http://baseUrl/spaces/127fd3a9-e071-4af8-b05f-de2b65887762",
              "iconClass": "icon-file-image",
              "isRecentResult": false,
              "lastModified": "2019-10-28T23:35:34.642Z",
              "name": "delectus.jpg",
              "resultId": "127fd3a9-e071-4af8-b05f-de2b65887762",
              "resultType": "confluence-object-result",
              "spaceId": "8bc813cd-632b-4bb8-8725-42bacb7ec3d2",
            },
          ],
          "timings": 0,
          "totalSize": 99,
        },
      }
    }
    people={
      Object {
        "isLoading": true,
        "results": null,
      }
    }
    query="query"
    screenType="cachedResults"
    spaces={
      Object {
        "isLoading": true,
        "results": null,
      }
    }
  />
  <LoadingSpinner />
</Fragment>
`;

exports[`<SearchResults /> render with results matches snapshot for post query 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    isCollapsed={false}
    isLoading={false}
    items={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-confluence",
              "containerId": "UNAVAILABLE",
              "containerName": "Lebsack - Hirthe",
              "contentType": "confluence-blogpost",
              "friendlyLastModified": "about 7 hours ago",
              "href": "http://baseUrl/spaces/cf94aabc-36e8-4a29-a807-7faa28c92792",
              "iconClass": "aui-iconfont-page-default",
              "isRecentResult": false,
              "lastModified": "2019-07-08T02:54:38.822Z",
              "name": "occaecati et consequatur amet dicta",
              "resultId": "cf94aabc-36e8-4a29-a807-7faa28c92792",
              "resultType": "confluence-object-result",
              "spaceId": "06ee101a-d558-40c9-854a-e468f9abf979",
            },
          ],
          "timings": 0,
          "totalSize": 99,
        },
      }
    }
    people={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-person",
              "avatarUrl": "http://lorempixel.com/640/480/people",
              "contentType": "person",
              "href": "/people/cbe55157-437c-4e20-8569-106485b4a3df",
              "mentionName": "nickname-ri83n",
              "name": "Madyson Lubowitz",
              "presenceMessage": "",
              "resultId": "people-cbe55157-437c-4e20-8569-106485b4a3df",
              "resultType": "person-result",
              "userId": "cbe55157-437c-4e20-8569-106485b4a3df",
            },
          ],
          "timings": 0,
        },
      }
    }
    query="query"
    screenType="postQuerySearchResults"
    spaces={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-confluence",
              "avatarUrl": "http://lorempixel.com/640/480/cats",
              "contentType": "confluence-space",
              "href": "http://baseUrl/spaces/REDEFINE",
              "id": "71629322-521e-4b19-89de-4c75c82ec38a",
              "key": "REDEFINE",
              "name": "Spinka and Sons",
              "resultId": "space-REDEFINE",
              "resultType": "generic-container-result",
            },
          ],
          "timings": 0,
        },
      }
    }
  />
</Fragment>
`;

exports[`<SearchResults /> render with results matches snapshot for post query with filter on 1`] = `
<Fragment>
  <WithAnalyticsEvents(WrappedComponent)
    isCollapsed={true}
    isLoading={false}
    items={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-confluence",
              "containerId": "UNAVAILABLE",
              "containerName": "Konopelski LLC",
              "contentType": "confluence-page",
              "friendlyLastModified": "Jan 23, 2018",
              "href": "http://baseUrl/spaces/1132af03-2630-4d26-858a-8ec56811db6a",
              "iconClass": "aui-iconfont-page-default",
              "isRecentResult": false,
              "lastModified": "2019-10-28T23:35:34.642Z",
              "name": "sapiente consequatur esse unde ea",
              "resultId": "1132af03-2630-4d26-858a-8ec56811db6a",
              "resultType": "confluence-object-result",
              "spaceId": "b61de77e-6195-47ae-92e6-7e19477f3c1b",
            },
          ],
          "timings": 0,
          "totalSize": 99,
        },
      }
    }
    people={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-person",
              "avatarUrl": "http://lorempixel.com/640/480/people",
              "contentType": "person",
              "href": "/people/5aff4d5d-be06-42ec-b67c-0922d664e88b",
              "mentionName": "nickname-1iepp",
              "name": "Ada DuBuque",
              "presenceMessage": "",
              "resultId": "people-5aff4d5d-be06-42ec-b67c-0922d664e88b",
              "resultType": "person-result",
              "userId": "5aff4d5d-be06-42ec-b67c-0922d664e88b",
            },
          ],
          "timings": 0,
        },
      }
    }
    query="query"
    screenType="postQuerySearchResults"
    spaces={
      Object {
        "isLoading": false,
        "results": Object {
          "items": Array [
            Object {
              "analyticsType": "result-confluence",
              "avatarUrl": "http://lorempixel.com/640/480/cats",
              "contentType": "confluence-space",
              "href": "http://baseUrl/spaces/PROGRESSIVE",
              "id": "1d0cbb4e-0c1e-4361-bb36-71511036ae53",
              "key": "PROGRESSIVE",
              "name": "Hyatt - Gorczany",
              "resultId": "space-PROGRESSIVE",
              "resultType": "generic-container-result",
            },
          ],
          "timings": 0,
        },
      }
    }
  />
</Fragment>
`;

exports[`<SearchResults /> render with results matches snapshot for pre query 1`] = `
<WithAnalyticsEvents(WrappedComponent)
  isCollapsed={false}
  isLoading={false}
  items={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-confluence",
            "containerId": "UNAVAILABLE",
            "containerName": "Hilpert Group",
            "contentType": "confluence-page",
            "friendlyLastModified": "about 7 hours ago",
            "href": "http://baseUrl/spaces/a430a73d-9e9b-4cce-86a5-1618965ffea6",
            "iconClass": "aui-iconfont-page-default",
            "isRecentResult": false,
            "lastModified": "2019-07-08T02:54:38.822Z",
            "name": "soluta non omnis vel ut",
            "resultId": "a430a73d-9e9b-4cce-86a5-1618965ffea6",
            "resultType": "confluence-object-result",
            "spaceId": "23683393-e090-4492-bd03-e808d9d81098",
          },
        ],
        "timings": 0,
        "totalSize": 99,
      },
    }
  }
  people={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-person",
            "avatarUrl": "http://lorempixel.com/640/480/people",
            "contentType": "person",
            "href": "/people/40f67ee1-7558-4f56-b601-16bcd266da80",
            "mentionName": "nickname-cnq03",
            "name": "Miss Agnes Harvey",
            "presenceMessage": "",
            "resultId": "people-40f67ee1-7558-4f56-b601-16bcd266da80",
            "resultType": "person-result",
            "userId": "40f67ee1-7558-4f56-b601-16bcd266da80",
          },
        ],
        "timings": 0,
      },
    }
  }
  spaces={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-confluence",
            "avatarUrl": "http://lorempixel.com/640/480/cats",
            "contentType": "confluence-space",
            "href": "http://baseUrl/spaces/PROGRAMMING",
            "id": "ec8184c9-4bf7-48c1-b98f-8e79c41fa450",
            "key": "PROGRAMMING",
            "name": "Padberg - Kirlin",
            "resultId": "space-PROGRAMMING",
            "resultType": "generic-container-result",
          },
        ],
        "timings": 0,
      },
    }
  }
/>
`;

exports[`<SearchResults /> render with results matches snapshot for pre query loading 1`] = `<LoadingSpinner />`;

exports[`<SearchResults /> render with results matches snapshot for pre query with filter on 1`] = `
<WithAnalyticsEvents(WrappedComponent)
  isCollapsed={false}
  isLoading={false}
  items={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-confluence",
            "containerId": "UNAVAILABLE",
            "containerName": "Russel, Grady and Baumbach",
            "contentType": "confluence-attachment",
            "friendlyLastModified": "Dec 23, 2018",
            "href": "http://baseUrl/spaces/f4042cad-cad4-4806-a18d-01ced82d4e59",
            "iconClass": "dummy-unmatched-icon-class",
            "isRecentResult": false,
            "lastModified": "2019-07-08T02:54:38.822Z",
            "name": "eos.test",
            "resultId": "f4042cad-cad4-4806-a18d-01ced82d4e59",
            "resultType": "confluence-object-result",
            "spaceId": "4f733a07-2edb-4737-8bb8-c879cb27b158",
          },
        ],
        "timings": 0,
        "totalSize": 99,
      },
    }
  }
  people={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-person",
            "avatarUrl": "http://lorempixel.com/640/480/people",
            "contentType": "person",
            "href": "/people/ea1039c0-1e13-457d-81ca-fa1cf4cc8df5",
            "mentionName": "nickname-nehql",
            "name": "Mya Rau",
            "presenceMessage": "",
            "resultId": "people-ea1039c0-1e13-457d-81ca-fa1cf4cc8df5",
            "resultType": "person-result",
            "userId": "ea1039c0-1e13-457d-81ca-fa1cf4cc8df5",
          },
        ],
        "timings": 0,
      },
    }
  }
  spaces={
    Object {
      "isLoading": false,
      "results": Object {
        "items": Array [
          Object {
            "analyticsType": "result-confluence",
            "avatarUrl": "http://lorempixel.com/640/480/cats",
            "contentType": "confluence-space",
            "href": "http://baseUrl/spaces/VIRGIN ISLANDS, U.S.",
            "id": "4a56bc68-196e-48d1-bd38-d97d60ec9cfc",
            "key": "VIRGIN ISLANDS, U.S.",
            "name": "Buckridge - Jacobi",
            "resultId": "space-VIRGIN ISLANDS, U.S.",
            "resultType": "generic-container-result",
          },
        ],
        "timings": 0,
      },
    }
  }
/>
`;

exports[`<SearchResults /> with no results render and matches snapshot for post query loading 1`] = `
<Fragment>
  <LoadingSpinner />
</Fragment>
`;
